### PR TITLE
Issue with show lldp neighbors and long names fixes #103

### LINKED ISSUE #86: get_lldp_neighbors fail if capabilities not present
### Description of Issue/Question
get_lldp_neighbors failed if device line does not have capabilities. maybe we should implement the case in textfsm and handle it?

### Did you follow the steps from https://github.com/napalm-automation/napalm#faq
- [x ] Yes
- [ ] No

### napalm-ios version
(Paste verbatim output from `pip freeze | grep napalm-ios` between quotes below)

```
napalm-ios==0.4.0
```

### IOS version
(Paste verbatim output from `show version` between quotes below)

```
Cisco IOS Software, IOS-XE Software, Catalyst 4500 L3 Switch Software (cat4500es8-UNIVERSALK9-M), Version 03.03.02.XO RELEASE SOFTWARE (fc1)
Technical Support: http://www.cisco.com/techsupport
Copyright (c) 1986-2014 by Cisco Systems, Inc.
Compiled Tue 19-Aug-14 02:17 by prod_rel_team



Cisco IOS-XE software, Copyright (c) 2005-2013 by cisco Systems, Inc.
All rights reserved.  Certain components of Cisco IOS-XE software are
licensed under the GNU General Public License ("GPL") Version 2.0.  The
software code licensed under GPL Version 2.0 is free software that comes
with ABSOLUTELY NO WARRANTY.  You can redistribute and/or modify such
GPL code under the terms of GPL Version 2.0.  For more details, see the
documentation or "License Notice" file accompanying the IOS-XE software,
or the applicable URL provided on the flyer accompanying the IOS-XE
software.



ROM: 15.1(1r)SG2

```

### Steps to Reproduce the Issue

Connect a security camera, dell server or any simple device that does not exchange capabilities. Then call get_lldp_neighbors()

### Error Traceback
(Paste the complete traceback of the exception between quotes below)

```
  File "/devs/lib/python2.7/site-packages/napalm_ios/ios.py", line 415, in get_lldp_neighbors
    device_id, local_int_brief, hold_time, capability, remote_port = lldp_entry.split()
ValueError: need more than 4 values to unpack
```

### LINKED ISSUE #103: get_lldp_neighbors() fails with long neighbor names
### Description of Issue/Question
get_lldp_neighbors() seems to depend on spaces to separate fields of the "show lldp neighbors" command.
In some cases, at least that IOS version, the output can have the columns "touching" each others, especially for neighbor with long host-names.

Device ID           Local Intf     Hold-time  Capability      Port ID
foo.foobar.foo   Gi2/0/2        121        O,R             ethernet1/16
foobar.foobar.foobarGi2/0/4        120        B               Gi2/1/2

Reducing the number of columns to 4 (see stack-trace bellow).

Note that the same issue happen if I explicitly set the terminal to the max width. 
#terminal width ?
  <0-512>  Number of characters on a screen line.

### Did you follow the steps from https://github.com/napalm-automation/napalm#faq
- [x] Yes
- [ ] No

### napalm-ios version
(Paste verbatim output from `pip freeze | grep napalm-ios` between quotes below)
```
napalm-ios==0.5.0
```
### IOS version
(Paste verbatim output from `show version` between quotes below)
```
Cisco IOS Software, IOS-XE Software, Catalyst L3 Switch Software (CAT3K_CAA-UNIVERSALK9-M), Version 03.06.04.E RELEASE SOFTWARE (fc2)
```
### Error Traceback
(Paste the complete traceback of the exception between quotes below)
```
>>> device. get_lldp_neighbors()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/xionox/Documents/Projects/Mozilla/network-automation/env/local/lib/python2.7/site-packages/napalm_ios/ios.py", line 453, in get_lldp_neighbors
    device_id, local_int_brief, hold_time, capability, remote_port = lldp_entry.split()
ValueError: need more than 4 values to unpack

```

### PR DESCRIPTION
**Pull request by ktbyers**
Also fixes #86

### CODE REVIEW

**Comment by ktbyers** [comment]
Needs to be tested on real device by @XioNoX

**Review by mirceaulinic — APPROVED** [review]

**Comment by XioNoX** [comment]
Works fine for me!

**Comment by ktbyers** [comment]
@mirceaulinic Can you review this second commit also (as there was a separate bug on 'show lldp neighbors' where the capability field was missing). That was a pretty simple fix.

File: napalm_ios/ios.py [hunk 1]
```diff
@@ -582,19 +582,24 @@ def get_lldp_neighbors(self):
...
-                    device_id = tmp_field[:20]
-                    local_int_brief = tmp_field[20:]
-                    # device_id might be abbreviated, try to get full name
-                    lldp_tmp = self._lldp_detail_parser(local_int_brief)
-                    device_id_new = lldp_tmp[3][0]
-                    # Verify abbreviated and full name are consistent
-                    if device_id_new[:20] == device_id:
-                        device_id = device_id_new
+                    # Four fields might be long_name or missing capability
+                    capability_missing = True if lldp_entry[46] == ' ' else False
```
Comment (mirceaulinic) [inline napalm_ios/ios.py]: I believe this is correct, but I'm just wondering: why 46? :)

**Review by mirceaulinic — APPROVED** [review]
Looks good!

**Comment by XioNoX** [comment]
2nd commit works as well for me!

File: napalm_ios/ios.py (hunk 1 above)
Comment (ktbyers) [reply napalm_ios/ios.py]: Short answer...   F****ing Cisco

Longer answer...Cisco merged two fields together (no spaces between them) so it is really hard to parse. Looking at the output, it looks like the field widths are fixed number of spaces apart and the 'capability' field starts at 47th character (i.e. index 46). If there is nothing, there then the capability is missing.

Comment (mirceaulinic) [reply napalm_ios/ios.py]: (facepalm)

Thanks for info!
